backend/dataall/cdkproxy/stacks/policies/_lambda.py

```diff
@@ -30,5 +30,18 @@ def get_statements(self):
                     f'arn:aws:lambda:{self.region}:{self.account}:layer:{self.resource_prefix}*:*',
                 ],
```

Comment:
Could we add a condition similar to other policies here

```
                conditions={
                    'StringEquals': {
                        f'aws:resourceTag/{self.tag_key}': [self.tag_value]
                    }
                },
```

Reply:
Yes we can (https://docs.aws.amazon.com/service-authorization/latest/reference/list_awslambda.html#awslambda-policy-keys) let me add it at least to the functions. Good catch

Reply:
I think we can further enhance the policies. Some only include conditions on `aws:ResourceTag`, but users can modify the tag of their resources once they are created. To ensure that they can only tag resources with their own tag we need to add a `aws:RequestTag` condition in those policies where is not already there